### PR TITLE
zbar: add patches for CVE-2023-40889 & CVE-2023-40890

### PR DESCRIPTION
**Pull request by risicle**
## Description of changes
https://nvd.nist.gov/vuln/detail/CVE-2023-40889
https://nvd.nist.gov/vuln/detail/CVE-2023-40890

Unfixed upstream, these are both (based on) third party patches. Can't test against the PoC input because the reporter didn't include that, but each patch addresses a lack of bounds checking at approximately the location of the reporter's stack trace.

On macos 10.15 & nixos aarch64-linux have built just `.*zbar.*` packages.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/zbar/default.nix [hunk 1]
```diff
@@ -1,6 +1,7 @@
 { stdenv
 , lib
 , fetchFromGitHub
+, fetchpatch
```
Comment (delroth) [inline pkgs/tools/graphics/zbar/default.nix:4]: Can be removed / not added.

**Review by delroth — APPROVED** [review]

File: pkgs/tools/graphics/zbar/default.nix (hunk 1 above)
Comment (7c6f434c) [reply pkgs/tools/graphics/zbar/default.nix:4]: Will probably need it when upstream adds more patches but takes some time to do the release (I am moderately optimistic, I guess)…

Comment (risicle) [reply pkgs/tools/graphics/zbar/default.nix:4]: Damnit I'm always doing this.

**Comment by risicle** [comment]
https://github.com/mchehab/zbar/issues/263 upstream

**Pull request merged**